Rater.Data/Repositories/Decorator/CachedRatingRepository.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.Extensions.Caching.Memory;
+using Rater.API;
+using Rater.Data.Repositories.Interfaces;
+using Rater.Domain.DataTransferObjects.RatingDto;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Rater.Data.Repositories.Decorator
+{
+    public class CachedRatingRepository : IRatingRepository
+    {
+        private readonly RatingRepository _decorated;
+        private readonly IMemoryCache _memoryCache;
+        public CachedRatingRepository(RatingRepository decorated , IMemoryCache memoryCache)
+        {
+            _decorated = decorated;
+            _memoryCache = memoryCache;
+        }
+
+        public async Task<RatingResponseDto> AddRatings(List<Rating> request)
+        {
+            return await _decorated.AddRatings(request);
```

Comment:
We should considering more feasible way to validate/invalidate cache status, I don't think giving a TTL time really make sense in this context because if you put 10 mins in there, what will happen, will everyone see the same results for 10 mins even there's an update?

Let's give a TTL for sure and if an user submits ratings, delete the ratings cache of that space (if exist).